opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java

```diff
@@ -154,74 +155,87 @@ private void reduceEvent(OnmsEvent e, OnmsAlarm alarm, Event event) {
         // Always set these
         alarm.setLastEvent(e);
         alarm.setLastEventTime(e.getEventTime());
-        alarm.setCounter(alarm.getCounter() + 1);
 
-        if (!event.getAlarmData().hasUpdateFields()) {
+        AlarmData alarmData = event.getAlarmData();
+        if (alarmData != null) {
```

Comment:
The event will be dropped earlier if this is null, so there's no need to check this again here - see  `checkEventSanityAndDoWeProcess`

Reply:
But asked in previous code change to do Null checking on these same objects.

Reply:
I had recommended changing the line in question to `if (Objects.equals(event.getAlarmData().getAlarmType(), alarm.getAlarmType())) {`.

Using `Objects.equals` in this case would only help prevent an NPE if/when the first call to `getAlarmType()` returned a null value.